addons/mail/models/mail_thread.py

```diff
@@ -668,44 +673,64 @@ def _message_track(self, fields_iter, initial_values_dict):
+                res = record._mail_track(tracked_fields, initial_values_dict[record.id])
+                tracking[record.id] = res
+                if res and res[0]:
+                    fields_changed.update(res[0])
             except MissingError:
                 continue
 
         # find content to log as body
         bodies = self.env.cr.precommit.data.pop(f'mail.tracking.message.{self._name}', {})
         authors = self.env.cr.precommit.data.pop(f'mail.tracking.author.{self._name}', {})
+        extra_subtypes = self.env["mail.message.subtype"]
+        if fields_changed:
+            extra_subtypes = extra_subtypes._get_matching_fields_tracked_subtypes(
+                self, list(fields_changed)
+            )
         for record in self:
             changes, tracking_value_ids = tracking.get(record.id, (None, None))
             if not changes:
                 continue
 
             # find subtypes and post messages or log if no subtype found
-            subtype = record._track_subtype(
-                dict((col_name, initial_values_dict[record.id][col_name])
-                     for col_name in changes)
-            )
+            initial_values = {
+                col_name: initial_values_dict[record.id][col_name] for col_name in changes
+            }
+            subtype = record._track_subtype(initial_values)
+            extra_subtypes_record = record._track_extra_subtypes(extra_subtypes, initial_values)
             author_id = authors[record.id].id if record.id in authors else None
             # _set_log_message takes priority over _track_get_default_log_message even if it's an empty string
             body = bodies[record.id] if record.id in bodies else record._track_get_default_log_message(changes)
+            message = None
             if subtype:
                 if not subtype.exists():
                     _logger.debug('subtype "%s" not found' % subtype.name)
                     continue
-                record.message_post(
+                message = record.message_post(
                     body=body,
                     author_id=author_id,
                     subtype_id=subtype.id,
                     tracking_value_ids=tracking_value_ids
                 )
             elif tracking_value_ids:
-                record._message_log(
+                message = record._message_log(
                     body=body,
                     author_id=author_id,
                     tracking_value_ids=tracking_value_ids
                 )
+            if extra_subtypes_record:
+                for extra_subtype in extra_subtypes_record:
+                    record.message_post(
+                        body=body,
+                        author_id=author_id,
+                        parent_id=message.id if message else None,
+                        subtype_id=extra_subtype.id,
+                        tracking_value_ids=tracking_value_ids,
+                    )
```

Comment:
Means doing all the work every single time and having copies of essentially the same message.
Should use message_notify instead and have 1 message.

Also the body is the same regardless of what I subscribe to? Then maybe consider supporting messages with multiple subtypes instead. Otherwise if you follow overlapping subtypes you'll get notified twice, I believe (maybe message_notify takes that into account based on existing notifications, not too sure)